DOC: correctly describe output shape of dirichlet.mean() and dirichlet.var()

cdouglass opened the pull request:
Fixes #7444. Also updates test to check exact output, including shape, since the existing test was consistent with the incorrect docstring info. Also did the same for the similar situation of `dirichlet.mean()`.

rgommers commented on scipy/stats/_multivariate.py:
```diff
@@ -1421,8 +1421,9 @@ def mean(self, alpha):
         -------
-        mu : scalar
-            Mean of the Dirichlet distribution
+        mu : ndarray
```
It's both:

    In [48]: stats.dirichlet.mean(alpha=[1])
    Out[48]: 1.0

    In [49]: stats.dirichlet.mean(alpha=[1, 2])
    Out[49]: array([ 0.33333333,  0.66666667])

I suggest making it ``scalar or ndarray``.

rgommers replied:
Annoyingly ``var`` doesn't go through ``_squeeze_output``, so does return an array of size 1. I suggest fixing that up so it does return a scalar.

rgommers replied:
Realize now that that was the original bug report - rather than changing the doc I think that ``var`` should be made consistent with the other methods. It's a very minor backwards compat break, but should be acceptable (needs a mention in the 1.0 release notes) imho.

rgommers commented:
It looks like there is no test for ``alpha`` of length-1 input, would be useful to add to ensure the return stays scalar.

cdouglass commented:
@rgommers I made the changes and added a test that the return value is scalar. I also noticed that several other methods in this module use `_squeeze_output` to give either a scalar or ndarray, so I added tests that `dirichlet.pdf` and `dirichlet.logpdf` return scalars and updated the docstrings for some other methods that already had such tests, such as `multivariate_normal.cdf`.

rgommers commented on doc/release/1.0.0-notes.rst:
```diff
@@ -75,6 +75,9 @@ has been changed to True.
 
 `scipy.special.gammaln` does not accept complex arguments anymore.
 
+The method `var` for the Dirichlet distribution of `scipy.stats.multivariate`
```
nitpick: ``var`` needs double backticks. single ones mean a link, and this link won't resolve. double backticks are for formatting as code

cdouglass replied:
Thanks, amended.

rgommers commented:
Thanks @cdouglass, all looks good to me now. Just had one very minor comment, but thought I'd make it anyway since this PR needs a rebase.

rgommers commented:
Rebased and merged in 415b74f. Thanks @cdouglass!
